### PR TITLE
add a screenshot command

### LINKED ISSUE #1004: add the ability to take a screenshot from the flutter tool
Is something like `flutter screenshot` in scope? I've needed to take a screenshot of the running app a few times, and always had to google how to do it with adb (`adb screencap` and then some file copying).

`flutter pic`? `flutter capture`?

Related to https://github.com/dart-atom/dartlang/issues/659.

### PR DESCRIPTION
**Pull request by devoncarew**
- add a screenshot command
- fix https://github.com/flutter/flutter/issues/1004, some work towards #2908 

@chinmaygarde, thanks for the ios simulator applescript! It does bring the simulator app to the foreground however. Is this something we can avoid or is it inevitable?

### CODE REVIEW

**Comment by sethladd** [comment]
Neat!

**Comment by chinmaygarde** [comment]
lgtm. We could probably tell which application is originally in the foreground and give focus back to it when done.

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -436,6 +436,20 @@ class AndroidDevice extends Device {
     runCheckedSync(cmd);
     return true;
   }
+
+  @override
+  bool get supportsScreenshot => true;
+
+  @override
+  Future<bool> takeScreenshot(File outputFile) {
+    const String remotePath = '/data/local/tmp/screen.png';
```
Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Might want to make this flutter-specific or random, but this is also OK.  does /data/local/tmp always exist?

Comment (abarth) [inline packages/flutter_tools/lib/src/android/android_device.dart]: We already rely on `/data/local/tmp` existing and being read/write by the shell user in order to launch apps.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Yup, I borrowed this path from some existing bits of our adb code; I updated the file name we put in there to `flutter_screenshot.png`.

**Pull request merged**